pkg/cloudprovider/vsphere/cloud.go

```diff
@@ -161,31 +174,30 @@ func (vs *VSphere) HasClusterID() bool {
 }
 
 // Initializes vSphere from vSphere CloudProvider Configuration
-func buildVSphereFromConfig(cfg *CPIConfig) (*VSphere, error) {
+func buildVSphereFromConfig(cfg *ccfg.CPIConfig, lbcfg *lcfg.LBConfig) (*VSphere, error) {
 	nm := newNodeManager(cfg, nil)
-	lb, err := loadbalancer.NewLBProvider(&cfg.LBConfig)
+
+	lb, err := loadbalancer.NewLBProvider(lbcfg)
 	if err != nil {
 		return nil, err
 	}
-	if _, ok := os.LookupEnv("ENABLE_ALPHA_NSXT_LB"); !ok {
-		if lb != nil {
-			klog.Infof("To enable NSX-T load balancer support you need to set the env variable ENABLE_ALPHA_NSXT_LB")
-			lb = nil
-		}
-	} else {
+	if _, ok := os.LookupEnv("ENABLE_ALPHA_NSXT_LB"); ok {
 		if lb == nil {
-			return nil, fmt.Errorf("To enable NSX-T load balancer support you need to configure section LoadBalancer")
+			klog.Warning("To enable NSX-T load balancer support you need to configure section LoadBalancer")
+		} else {
+			klog.Infof("NSX-T load balancer support enabled. This feature is alpha, use in production at your own risk.")
+			// redirect vapi logging from the NSX-T GO SDK to klog
+			log.SetLogger(NewKlogBridge())
 		}
-	}
-	if lb == nil {
-		klog.Infof("NSX-T load balancer support disabled")
 	} else {
-		klog.Infof("NSX-T load balancer support enabled. This feature is alpha, use in production at your own risk.")
-		// redirect vapi logging from the NSX-T GO SDK to klog
-		log.SetLogger(NewKlogBridge())
+		// explicitly nil the LB interface if ENABLE_ALPHA_NSXT_LB is not set even if the LBConfig is valid
```

Comment:
Previous this was 
```	
if lb != nil {
    klog.Infof("To enable NSX-T load balancer support you need to set the env variable ENABLE_ALPHA_NSXT_LB")
    lb = nil
}
```

Do we want to keep that log message?

Reply:
To formatting is a little weird... it should look like this:

`	if _, ok := os.LookupEnv("ENABLE_ALPHA_NSXT_LB"); ok {
		if lb == nil {
			klog.Warning("To enable NSX-T load balancer support you need to configure section LoadBalancer")
		} else {
			klog.Infof("NSX-T load balancer support enabled. This feature is alpha, use in production at your own risk.")
			// redirect vapi logging from the NSX-T GO SDK to klog
			log.SetLogger(NewKlogBridge())
		}
	} else {
		// explicitly nil the LB interface if ENABLE_ALPHA_NSXT_LB is not set even if the LBConfig is valid
		// ENABLE_ALPHA_NSXT_LB must be explicitly enabled
		lb = nil
	}`

Reply:
So basically if someone has the ENV set, and `lb == nil`, then the interface isn't set and we should warn the user.
If both are set, we fall into the else.

The problem with outer else... is someone didn't set the ENVVAR but the interface isn't `nil` and it would cause the CCM to crash.